process/coordinator/process.go

```diff
@@ -1401,19 +1435,25 @@ func (tc *transactionCoordinator) getAllTransactions(body *block.Body) map[block
 }
 
 func (tc *transactionCoordinator) verifyGasLimit(
+	header data.HeaderHandler,
 	body *block.Body,
 	mapMiniBlockTypeAllTxs map[block.Type]map[string]data.TransactionHandler,
 ) error {
-	for _, miniBlock := range body.MiniBlocks {
+	for index, miniBlock := range body.MiniBlocks {
 		isCrossShardMiniBlockFromMe := miniBlock.SenderShardID == tc.shardCoordinator.SelfId() &&
 			miniBlock.ReceiverShardID != tc.shardCoordinator.SelfId()
 		if !isCrossShardMiniBlockFromMe {
 			continue
 		}
-
 		if miniBlock.Type == block.SmartContractResultBlock {
 			continue
 		}
+		if tc.flagScheduledMiniBlocks.IsSet() {
+			if tc.scheduledTxsExecutionHandler.IsMiniBlockExecuted(header.GetMiniBlockHeadersHashes()[index]) {
```

Comment:
Yep, this will be done by taking it from the mini block header IsFinal property. I will add also here a TODO to be refactored after we have the last version in the elrond-go-core repository

Reply:
Actually we can remove this only when we could differentiate between the final mini blocks executed as scheduled in the last block and the normal mini blocks from the current block which are also final, but not yet executed